libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -4175,13 +4178,13 @@ string YulUtilFunctions::tryDecodeErrorMessageFunction()
 				let end := add(add(msg, 0x20), length)
 				if gt(end, add(data, sub(returndatasize(), 4))) { leave }
 
-				mstore(<freeMemoryPointer>, add(add(msg, 0x20), <roundUp>(length)))
+				<finalizeAllocation>(data, add(offset, add(0x20, length)))
```

Comment:
Not sure why we are replacing `mstore(freeMemPtr, msg + 0x20 + round(length))` by `mstore(data + round(offset + 0x20 + length)`

Reply:
Because `finalize_allocation()` now manages `freeMemPtr`. The code is equivalent because `msg` is equal to `data + offset`, and `finalize_allocation()` already does `roundUp()`. The only difference is that `finalize_allocation()` also checks that the allocated memory area isn't too large (which probably accounts for some of the increased gas usage).